Hubspot doc refactor (#907)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/hubspot/hubspot-cloud-mode/hubspot-new-api-cloud.mdx

---
title: "HubSpot Cloud Mode (New API)"
description: Detailed technical documentation on sending events to HubSpot via cloud mode using the new API.
---

RudderStack supports the following API calls while sending data to Hubspot via <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link> using the [new API](https://developers.hubspot.com/docs/api/overview).

## Identify

RudderStack sends the <Link to="/event-spec/standard-events/identify">`identify`</Link> call to check if a contact exists, if it does, RudderStack updates the contact with new information. Otherwise, it creates a new contact.

A sample `identify` call is as shown:

```javascript
rudderanalytics.identify({
  firstName: "Alex",
  city: "New Orleans",
  country: "USA",
  phone: "+1-202-555-0146",
  email: "alex@example.com",
  custom_flavor: "chocolate",
  custom_date: 1574769933368,
  custom_date1: new Date("2019-10-14T11:15:53.296Z"),
})

> ```

Reviewer: Same comment as [here](https://github.com/rudderlabs/rudderstack-docs/pull/907/files#r986625512)

Reviewer 2: This can be for demonstration purposes like externalId parameter [above](https://github.com/rudderlabs/rudderstack-docs/pull/907#discussion_r986630068).

> The lookup field passed in the **HubSpot property name to be used as lookup field** dashboard setting must also be passed in the `identify` call's `traits`. For example, if you have passed `sampleField` in the dashboard, you must pass it and its corresponding value in the `identify` call, as shown:

Reviewer 2: In the UI we have provided different example name as (`eg: lookupField`)

Reviewer 2: `sampleField` also showcases a general meaning that the name depends on the user

```javascript
rudderanalytics.identify({
  firstName: "Alex",
  sampleField: "phone",
  phone: "+1-202-555-0146"
})
```
If you choose to use the lookup field (`sampleField` in the above example), you can change the lookup value (`phone` in the above example) anytime just by passing the new value in the `identify` call without editing your dashboard setting.

<div class="infoBlock">
If you pass a lookup field in the dashboard setting but not under the <code class="inline-code">identify</code> call's <code class="inline-code">traits</code>, RudderStack  will take <code class="inline-code">email</code> as the default lookup field.
</div>

## Track

A <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you record various user actions and any properties associated with them. 

To associate a `track` call with a user, you need to specify the user's `email` under `context.traits`. 

<div class="infoBlock">

> RudderStack associates the <code class="inline-code">track</code> events with the same user after an <code class="inline-code">identify</code> request is successfully made. However, if you send a <code class="inline-code">track</code> call without making any <code class="inline-code">identify</code> call first, and there is no contact in HubSpot, there will be no association of events as there is no identifier.

Reviewer: ```suggestion
RudderStack associates the <code class="inline-code">track</code> events with the same user only after you make a successful <code class="inline-code">identify</code> call. However, if you send a <code class="inline-code">track</code> call without making any <code class="inline-code">identify</code> call first and there is no contact present in HubSpot, RudderStack will not associate the events as there is no identifier.
```
